cpp/arcticdb/column_store/column_data.hpp

```diff
@@ -133,6 +133,78 @@ struct ColumnData {
  */
 
   public:
+    template<typename TDT, bool constant>
+    class ColumnDataIterator: public boost::iterator_facade<
+            ColumnDataIterator<TDT, constant>,
+            std::conditional_t<constant, typename TDT::DataTypeTag::raw_type const, typename TDT::DataTypeTag::raw_type>,
+            boost::forward_traversal_tag
+            > {
+    using RawType = typename TDT::DataTypeTag::raw_type;
+    public:
+        ColumnDataIterator() = delete;
+
+        // Used to construct [c]begin iterators
+        explicit ColumnDataIterator(ColumnData* parent):
+        parent_(parent)
+        {
+            opt_block_ = parent->next<TDT>();
+            if(opt_block_.has_value()) {
+                remaining_values_in_block_ = opt_block_->row_count();
+                if constexpr(constant) {
```

Comment:
Isn't this redundant. It looks like we have already set the "const type" in the constructor
```c++
std::conditional_t<constant, typename TDT::DataTypeTag::raw_type const, typename TDT::DataTypeTag::raw_type>
```
can't we use `ColumnIterator::type` or whatever boost defined the result of the conditional to be?

Reply:
This one is needed, as it flips between `reinterpret_cast` and 'const_cast'.
I've updating the ``RawType definition to:
`using RawType = std::conditional_t<constant, const typename TDT::DataTypeTag::raw_type, typename TDT::DataTypeTag::raw_type>;`
though, which removes the need for a bunch of other `std::conditional_t`s